weblate/wladmin/sites.py

```diff
@@ -12,7 +12,7 @@
 from django.utils.translation import gettext, gettext_lazy
 
 if TYPE_CHECKING:
-    from weblate.auth.models import AuthenticatedHttpRequest
+    from django.http import HttpRequest
```

Comment:
The intention here is that Django's `HttpRequest` can have `AnonymousUser` what can't happen in Weblate. That's why `AuthenticatedHttpRequest` was introduced. Without that each access to `reuqest.user` has complains from mypy.

Reply:
Actually, this is an upstream recommended approach: https://github.com/typeddjango/django-stubs?tab=readme-ov-file#how-can-i-create-a-httprequest-thats-guaranteed-to-have-an-authenticated-user

Reply:
Yes, I've seen that answer in FAQ. The reason of this change is because of a method `each_context`, that has a signature with `HttpRequest`. The mypy error is:

error: Argument 1 of "each_context" is incompatible with supertype "AdminSite"; supertype defines the argument type as "HttpRequest"  [override]
note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
note: This violates the Liskov substitution principle

For now I have two solutions on my mind, `cast` and `ignore`.

* `cast("AuthenticatedHttpRequest", request)` doesn't make much sense, because the function doesn't actually need a `request.user`
* `# type: ignore[override]` doesn't seem that bad actually

Reply:
I was using ignore for override so far, IMHO it is the best approach